### ISSUE TITLE
LLE counter-current rate-based ODE has wrong sign for organic phase

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#82]
## Description

In src/difflow/units/lle.py (lines 862-880), the counter-current extraction model uses ODEs for both aqueous and organic phases. The organic phase flows counter to the aqueous phase, so its ODE should have opposite sign for the spatial derivative. The current implementation has the same sign for both phases, which means both phases are modeled as co-current flow.

In counter-current extraction:
- dC_aq/dz = -K_L * a * (C_aq - C_aq*) (aqueous flows in +z direction)
- dC_org/dz = +K_L * a * (C_aq - C_aq*) * (A_aq/A_org) (organic flows in -z direction, so positive dC/dz)

## Impact

The counter-current extraction model behaves as co-current, significantly underestimating extraction efficiency.

## Suggested Fix

Negate the organic phase ODE to account for counter-current flow direction.

## Severity
CRITICAL

## Component
Core - Separation (src/difflow/units/lle.py)

**Comment by jkitchin** [comment]
## Fixed in commit 0f395a1

### Root cause
In `DifferentialContactor._solve_ode()` (`src/difflow/units/lle.py`), the system matrix for the counter-current ODE had the wrong sign for the organic phase row, and the matrix construction used `.transpose(2,0,1)` which broke for `n_solutes=1`.

The old code had:
```python
# Wrong: both rows negative
A = [[-a,  b],
     [-c,  d]]   # <-- should be [c, -d]
```

In counter-current flow, the organic phase flows in the -z direction. At a differential element, the organic phase *gains* solute (c_org increases where organic exits enriched), so the ODE is:
```
dc_org/dz = +Kla*(K*c_aq - c_org)*area/F_org   (positive sign)
```

### Fix
1. Changed the organic row from `[-c, d]` to `[c, -d]`.
2. Replaced `.transpose(2,0,1)` with `jnp.stack([row0, row1], axis=1)` which correctly produces shape `(n_solutes, 2, 2)` for any number of solutes including 1.

```python
# Corrected matrix
row0 = jnp.stack([-a, b], axis=-1)   # (n_solutes, 2)
row1 = jnp.stack([c, -d], axis=-1)   # (n_solutes, 2)
A_matrices = jnp.stack([row0, row1], axis=1)  # (n_solutes, 2, 2)
```

### Tests
`tests/test_lle_distillation_bugs.py` -- verifies that the differential contactor produces physically correct extraction (organic phase enriched, aqueous phase depleted), mass balance closes, and the solution works for both single and multiple solutes.

**Comment by jkitchin** [comment]
Closed via commit 0f395a1.
